### PR TITLE
6771 pricefeed errors

### LINKED ISSUE #6771: apparent success in failed price aggregator submissions
## Describe the bug

In some cases when `PushPrice` fails there is no way for the client to detect it.

## To Reproduce

Three cases from @jacquesvcritien:

1) This is an incorrect input submissions and even though there is an error, you can still see `numWantsSatisfied: 1`
image

2) This is a submission when a previous round was already created by the same oracle. The logic is handled well in the contract because a new round isn’t created. However, you can still see `numWantsSatisfied: 1`

image (1)

3) I tried submitting a price to an old round. Once again, since we were already on round 3, this was rejected by the logic. However, you can still see `numWantsSatisfied: 1`

image (2)
image (3)

## Expected behavior
When there is an error, the `error` property is populated.

numWantsSatisfied should be one because the wants were trivially satisfied by there not being any :)

## Platform Environment
current master

### PR DESCRIPTION
**Pull request by turadg**
closes: #6771

## Description

When a `PushPrice` invitation executed, errors in the `pushPrice` call weren't caught or reported.

The reason is `pushPrice` was called as a floating promise. We have a way to lint floating promises but it's [too slow for CI](https://github.com/Agoric/agoric-sdk/issues/5788). Still this error should serve as a motivation to [resolve them before the next release](https://github.com/Agoric/agoric-sdk/issues/6000), to detect any other surprises.

### Security Considerations

No new boundaries.

### Documentation Considerations

Conforms better to expectation. Nothing new to document.

### Testing Considerations

New tests.

### CODE REVIEW

File: packages/inter-protocol/src/price/priceAggregatorChainlink.js [hunk 1]
```diff
@@ -219,10 +219,17 @@ export const start = async (zcf, privateArgs, baggage) => {
...
-              admin.pushPrice(result);
-            }, 'PushPrice');
+            return zcf.makeInvitation(
+              /** @param {ZCFSeat} cSeat */
+              cSeat =>
+                E.when(
+                  admin.pushPrice(result),
+                  () => cSeat.exit(),
+                  // @ts-expect-error faulty onrejected type
+                  err => cSeat.fail(err),
```
Comment (turadg) [inline packages/inter-protocol/src/price/priceAggregatorChainlink.js]: could Zoe fail automatically if the offerhandler throws?

File: packages/smart-wallet/src/offers.js [hunk 2]
```diff
@@ -112,6 +112,16 @@ export const makeOfferExecutor = ({
         );
         logger.info(id, 'seated');
 
+        // NB: the following four promises all resolve exactly when the seat exits.
```
Comment (turadg) [inline packages/smart-wallet/src/offers.js]: @Chris-Hibbert is this true in general? @dckc has a counter example https://github.com/Agoric/agoric-sdk/issues/6777#issuecomment-1379417692

File: packages/inter-protocol/src/price/priceAggregatorChainlink.js (hunk 1 above)
Comment (turadg) [reply packages/inter-protocol/src/price/priceAggregatorChainlink.js]: it does!

File: packages/inter-protocol/src/price/priceAggregatorChainlink.js [hunk 3]
```diff
@@ -219,10 +219,14 @@ export const start = async (zcf, privateArgs, baggage) => {
...
           PushPrice(result) {
-            return zcf.makeInvitation(cSeat => {
-              cSeat.exit();
-              admin.pushPrice(result);
-            }, 'PushPrice');
+            return zcf.makeInvitation(
+              /** @param {ZCFSeat} cSeat */
+              async cSeat => {
+                cSeat.exit();
+                await admin.pushPrice(result);
```
Comment (turadg) [inline packages/inter-protocol/src/price/priceAggregatorChainlink.js:226]: when this throws, `getOfferResult()` rejects

**Review by dckc — APPROVED** [review]
test coverage looks pretty good. yay!

File: packages/smart-wallet/src/offers.js (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/smart-wallet/src/offers.js]: The way I read @dckc's comment, it was saying that if the seat doesn't exit, then various things don't work. That's consistent with my expectation that `numWantsSatisfied()`,  `getFinalAllocation()`, and `getPayouts()` resolve at the same time that the exitSubscriber reports that it's done.  The comment above seems to include `getOfferResult()`, which @erights pointed out often resolves at a different time.

File: packages/inter-protocol/src/price/priceAggregatorChainlink.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/price/priceAggregatorChainlink.js:226]: that seems right. (Sorry if I misled you previously.) 

`getOfferResult()` is supposed to give you the return value of the offerHandler, so if it throws, `getOfferResult()` has nothing else to return.  The exitSubscriber should still report that the seat has exited, and numWantsSatisfied should be 1.

File: packages/inter-protocol/test/smartWallet/test-oracle-integration.js [hunk 4]
```diff
@@ -127,7 +186,7 @@ test('admin price', async t => {
     'priceAggregator',
   );
 
-  // The purse has the invitation to get the makers ///////////
+  // The purse has the invitation to get the makers
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/test/smartWallet/test-oracle-integration.js:189]: Does the reason you removed that apply to line 227 as well?

**Review by Chris-Hibbert — APPROVED** [review]

File: packages/inter-protocol/test/smartWallet/test-oracle-integration.js (hunk 4 above)
Comment (turadg) [reply packages/inter-protocol/test/smartWallet/test-oracle-integration.js:189]: Just cleanliness. Not worth another push.

**Pull request merged**